deployment/scripts/upgrade.sh

```diff
@@ -0,0 +1,370 @@
+    oc scale --replicas=0 dc $dc
+  done
+  for pod in $(oc get pods -l component=es -o jsonpath='{.items[*].metadata.name}'); do
+    waitForStop $pod "component=es"
+  done
+}
+
+function scaleUp() {
+
+  # scale up ES slowly -- we want to wait for each instance to recover its indices first
+  local elasticsearch_dc=(`oc get dc -l logging-infra=elasticsearch -o jsonpath='{.items[*].metadata.name}'`)
+  for dc in "${elasticsearch_dc[@]}"; do
+    oc scale --replicas=1 dc $dc
+
+    #pod=$(oc get pod -l deploymentconfig=$dc -o jsonpath="{.items[?(.metadata.labels.deploymentconfig==\"$dc\")].metadata.name}")
+    pod=$(getPodName $dc)
+    waitForStart $pod "deploymentconfig=$dc"
+
+    # the first one we start gets to be the master
+    checkESStarted $pod
+  done
+
+  local kibana_dc=(`oc get dc -l logging-infra=kibana -o jsonpath='{.items[*].metadata.name}'`)
+  for dc in "${kibana_dc[@]}"; do
+    oc scale --replicas=1 dc $dc
+
+    #pod=$(oc get pod -l deploymentconfig=$dc -o jsonpath="{.items[?(.metadata.labels.deploymentconfig==\"$dc\")].metadata.name}")
+    pod=$(getPodName $dc)
+    waitForStart $pod "deploymentconfig=$dc"
+    checkKibanaStarted $pod
+  done
+
+  # this should just be scaling up fluentd to whichever nodes it was already deployed on
+  # if $fluentd_nodes is empty then we can't redeploy to the same nodes -- we should tell users to scale up manually
+  if [[ -n "$fluentd_nodes" ]]; then
+    for node in "${fluentd_nodes[@]}"; do
+      oc label node $node "$fluentd_nodeselector"
+
+      #TODO: we should be making sure our pods get scheduled and get to 'Running' before we say things are done...
+    done
+  else
+    echo "No previously labeled nodes for Fluentd were found; please label nodes to deploy Fluentd"
+  fi
+}
+
+# This should be used for the uuid_migrate only...
+function scaleSingleES() {
+  local elasticsearch_dc=(`oc get dc -l logging-infra=elasticsearch -o jsonpath='{.items[*].metadata.name}'`)
+  for dc in "${elasticsearch_dc[0]}"; do
+    oc scale --replicas=1 dc $dc
+
+    pod=$(getPodName $dc)
+    waitForStart $pod "deploymentconfig=$dc"
+
+    # the first one we start gets to be the master
+    checkESStarted $pod
+  done
+}
+
+function updateImages() {
+
+  # create any missing imagestreams and then update them all
+  oc new-app logging-imagestream-template || : # these may fail if created independently; that's ok
+
+  # TODO: update these to fail if we can't import?
+  oc import-image logging-elasticsearch || :
+  oc import-image logging-fluentd || :
+  oc import-image logging-kibana || :
+  oc import-image logging-curator || :
+}
+
+# we can just specify to regenerate our certs
+function add_curator() {
+
+  echo "Adding curator..."
+  # generate the template
+  generate_curator_template
+  generate_curator
+}
+
+function fluentd_daemonset() {
+  echo "Updating Fluentd to use Daemonset..."
+
+  # Delete old deployment and create new one
+  generate_fluentd_template
```

Comment:
I'm having trouble recreating this with an origin base install.  Not sure if this is due to me testing going from an enterprise install to an origin one... i'll have to investigate this further